sdk/python/tests/compiler/testdata/uri_artifacts.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Pipeline DSL code for testing URI-based artifact passing."""
+from kfp import compiler
+from kfp import components
+from kfp import dsl
+
+
+# Patch to make the test result deterministic.
+class Coder:
+  def __init__(self, ):
+    self._code_id = 0
+
+  def get_code(self, ):
+    self._code_id += 1
+    return '{code:0{num_chars:}d}'.format(
+        code=self._code_id,
+        num_chars=dsl._for_loop.LoopArguments.NUM_CODE_CHARS)
+
+
+dsl.ParallelFor._get_unique_id_code = Coder().get_code
+
+
+write_to_gcs = components.load_component_from_text("""
+name: Write to GCS
+inputs:
+- {name: text, type: String, description: 'Content to be written to GCS'}
+outputs:
+- {name: output_gcs_path, type: GCSPath, description: 'GCS file path'}
+implementation:
+  container:
+    image: google/cloud-sdk:slim
+    command:
+    - sh
+    - -c
+    - |
+      set -e -x
+      echo "$0" | gsutil cp - "$1"
+    - {inputValue: text}
+    - {outputUri: output_gcs_path}
+""")
+
+read_from_gcs = components.load_component_from_text("""
+name: Read from GCS
+inputs:
+- {name: input_gcs_path, type: GCSPath, description: 'GCS file path'}
+implementation:
+  container:
+    image: google/cloud-sdk:slim
+    command:
+    - sh
+    - -c
+    - |
+      set -e -x
+      gsutil cat "$0"
+    - {inputUri: input_gcs_path}
```

Comment:
I'm confused. Are we passing a URI of a URI here?

`inputPath: SomeModel` - path to file containing some model
`inputPath: SomeUri` - path to file containing some URI
`inputUri: SomeModel` - URI to a blob containing some model
`inputUri: SomeUri` - path to a blob containing some URI

Reply:
You have a good point. If one does `inputUri: SomeUri`then I guess yes, but that's something we wish to advertise against. Instead, we want users to directly access the pointer to the actual data content.

I'll change the use case away from `GCSPath`